### DISCUSSION TITLE
How can I load RPF/CADRG files via code?

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1948] [Support Q&A] [answered]
How can I load RPF files via code? In the previos version I could easly load the rpf files as image layer with the following code. But I cannot load them in the same way with version 3.2.

        GDALOptions gnc;
	gnc.setDriver("gdal");
	gnc.url() = "D:/maps/cgnc";
	ImageLayer* gncLayer = new ImageLayer("GNC", gnc);
	gncLayer->setMinVisibleRange(420000);
	gncLayer->setMaxVisibleRange(1166000);
	mapNode->getMap()->addLayer(gncLayer);

Edit: I managed to load to rpf files one by one. I also reated a .vrt file. But while creating the the vrt file it gives me color error. What I do to fix this issue?

**Comment by gwaldron** [comment, marked as answer]
I was going to suggest a VRT, but it sounds like you already figured that out. :)
Cannot comment on your "color error" without more information.

**Reply by nebsar** [reply]
When I use gdalbuildvrt.exe to create .vrt file I get the below warnings. Maybe I need to use BGR color. I do not know.

000GK011.ON2 has different values than the first raster for some entries in the color table.
The end result might produce weird colors.
You're advised to pre-process your rasters with other tools, such as pct2rgb.py or gdal_translate -expand RGB
to operate gdalbuildvrt on RGB rasters instead

Thanks Glenn

**Comment by nebsar** [comment]
I handled the problem using .toc file. Thanks Glenn.
